x/lockup/keeper/lock.go

```diff
@@ -413,22 +413,53 @@ func (k Keeper) Lock(ctx sdk.Context, lock types.PeriodLock) error {
 	return nil
 }
 
+// splitLock splits a lock with the given amount, and stores split new lock to the state
+func (k Keeper) splitLock(ctx sdk.Context, lock types.PeriodLock, coins sdk.Coins) (types.PeriodLock, error) {
```

Comment:
lets add a sanity check to ensure lock is a bonded lock? (No splitting unbonding locks)

Reply:
Added!